### PR TITLE
[TACHYON-1184] Add an interface to all client option builders

### PR DESCRIPTION
**Pull request by shuhuai007**
Please review
https://tachyon.atlassian.net/browse/TACHYON-1184

### CODE REVIEW

**Comment by aaudiber** [comment]
ok to test

**Comment by aaudiber** [comment]
@shuhuai007 Thanks for the patch!

**Comment by aaudiber** [comment]
+1 LGTM

**Comment by gpang** [comment]
ok to test

File: clients/unshaded/src/main/java/tachyon/client/file/options/OptionsBuilder.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file.options;
+
+/**
+ * This interface should be implemented by all Tachyon client builders.
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/file/options/OptionsBuilder.java]: Actually, I think it is not client builder but option builders for client-side operations

File: clients/unshaded/src/main/java/tachyon/client/file/options/InStreamOptions.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 @PublicApi
 public final class InStreamOptions {
-  public static class Builder {
+  public static class Builder  implements OptionsBuilder<InStreamOptions> {
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/file/options/InStreamOptions.java]: There seems an extra space between `Builder` and `implements`,

Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/file/options/InStreamOptions.java]: also in the following few declarations.

**Comment by apc999** [comment]
@shuhuai007 thanks for the patch to improve the code readability! I left two minor comments. Would you take a look? Thanks!

**Comment by shuhuai007** [comment]
@apc999 
thanks for your review.
I have modified these changes, please review again.

**Comment by aaudiber** [comment]
@shuhuai007 Thanks for the improvements, it looks good to me. In the future feel free to make formatting fixes in a separate commit (instead of `commit --amend` + `git push -f`). That makes it a little easier for reviewers since they can just look at the latest commit.

**Comment by apc999** [comment]
+1 LGTM
Thanks for the patch @shuhuai007

**Pull request merged**